Wrong KameletBinding definition leads to operator panic

Adding a wrong definition such as:
```
  errorHandler:
    sink:
      ref:
        kind: Kamelet
        apiVersion: camel.apache.org/v1alpha1
        name: err-sink
```
leads to a camel k operator panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c7cd8d]

goroutine 555 [running]:
github.com/apache/camel-k/pkg/controller/kameletbinding.maybeErrorHandler(0x2281e96, {{0x26111c0, 0xc0002b3d70}, {0x268e200, 0xc000641380}, {0xc000e49140, 0xd}, {0x228a5fe, 0xa}})
	github.com/apache/camel-k/pkg/controller/kameletbinding/error_handler.go:38 +0x10d
github.com/apache/camel-k/pkg/controller/kameletbinding.CreateIntegrationFor({0x26111c0, 0xc0002b3d70}, {0x268e200, 0xc000641380}, 0xc000d4ab40)
	github.com/apache/camel-k/pkg/controller/kameletbinding/integration.go:112 +0x7f6
...
```
Any wrong definition should be treated and reported as an error without operator crash.

--- a/pkg/apis/camel/v1alpha1/error_handler_types_support.go
+++ b/pkg/apis/camel/v1alpha1/error_handler_types_support.go
@@ -19,6 +19,7 @@ package v1alpha1
 
 import (
 	"encoding/json"
+	"fmt"
 )
 
 // +kubebuilder:object:generate=false
@@ -28,6 +29,7 @@ type ErrorHandler interface {
 	Type() ErrorHandlerType
 	Endpoint() *Endpoint
 	Configuration() (map[string]interface{}, error)
+	Validate() error
 }
 
 // baseErrorHandler is the base used for the Error Handler hierarchy
@@ -47,6 +49,11 @@ func (e baseErrorHandler) Endpoint() *Endpoint {
 // Configuration --
 func (e baseErrorHandler) Configuration() (map[string]interface{}, error) {
 	return nil, nil
+}
+
+// Validate --
+func (e baseErrorHandler) Validate() error {
+	return nil
 }
 
 // ErrorHandlerNone --
@@ -125,4 +132,12 @@ func (e ErrorHandlerSink) Configuration() (map[string]interface{}, error) {
 	properties[ErrorHandlerAppPropertiesPrefix] = "#class:org.apache.camel.builder.DeadLetterChannelBuilder"
 
 	return properties, err
+}
+
+// Validate --
+func (e ErrorHandlerSink) Validate() error {
+	if e.DLCEndpoint == nil {
+		return fmt.Errorf("Missing endpoint in Error Handler Sink")
+	}
+	return nil
 }

--- a/pkg/controller/kameletbinding/error_handler.go
+++ b/pkg/controller/kameletbinding/error_handler.go
@@ -79,8 +79,10 @@ func parseErrorHandler(rawMessage v1alpha1.RawMessage) (v1alpha1.ErrorHandler, e
 			return nil, errors.Errorf("Unknown error handler type %s", errHandlType)
 		}
 
-		err := json.Unmarshal(errHandlValue, dst)
-		if err != nil {
+		if err = json.Unmarshal(errHandlValue, dst); err != nil {
+			return nil, err
+		}
+		if err = dst.Validate(); err != nil {
 			return nil, err
 		}
 

--- a/pkg/controller/kameletbinding/error_handler_test.go
+++ b/pkg/controller/kameletbinding/error_handler_test.go
@@ -18,7 +18,6 @@ limitations under the License.
 package kameletbinding
 
 import (
-	"fmt"
 	"testing"
 
 	"github.com/apache/camel-k/v2/pkg/apis/camel/v1alpha1"
@@ -64,7 +63,6 @@ func TestParseErrorHandlerLogWithParametersDoesSucceed(t *testing.T) {
 }
 
 func TestParseErrorHandlerSinkDoesSucceed(t *testing.T) {
-	fmt.Println("Test")
 	sinkErrorHandler, err := parseErrorHandler(
 		[]byte(`{"sink": {"endpoint": {"uri": "someUri"}}}`),
 	)
@@ -100,4 +98,12 @@ func TestParseErrorHandlerSinkWithParametersDoesSucceed(t *testing.T) {
 	assert.Equal(t, v1alpha1.ErrorHandlerRefDefaultName, parameters[v1alpha1.ErrorHandlerRefName])
 	assert.Equal(t, "value1", parameters["camel.beans.defaultErrorHandler.param1"])
 	assert.Equal(t, "value2", parameters["camel.beans.defaultErrorHandler.param2"])
+}
+
+func TestParseErrorHandlerSinkFail(t *testing.T) {
+	_, err := parseErrorHandler(
+		[]byte(`{"sink": {"ref": {"uri": "someUri"}}}`),
+	)
+	assert.NotNil(t, err)
+	assert.Equal(t, "Missing endpoint in Error Handler Sink", err.Error())
 }

